norddrop/src/device/mod.rs

```diff
@@ -558,3 +555,48 @@ fn prepare_transfer_files(
 
     Ok(files)
 }
+
+async fn open_database(
+    dbpath: &str,
+    events: &EventDispatcher,
+    logger: &slog::Logger,
+) -> Result<drop_storage::Storage> {
+    // Try opening the DB 3 times
+    let mut i = 0;
+    let openerr = loop {
+        match drop_storage::Storage::new(logger.clone(), dbpath).await {
+            Ok(storage) => return Ok(storage),
+            Err(err) => {
+                i += 1;
+                warn!(logger, "Failed to open DB: {err}, already tried {i} times",);
+                if i >= 3 {
+                    break err;
+                }
+            }
+        }
+    };
```

Comment:
Do we really suspect that opening the database might fail in a recoverable manner? This code looks fine so no harm to have it, just curious.

Reply:
Good question and the answer is I don't know. But removing the DB in case it is somehow recoverable would be a major offence I guess. So I just wanted to be safe

Reply:
That's a fair point, was just curious